portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,25 +32,62 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    ArrayList<String> micaylaMessages = new ArrayList<>();
-    micaylaMessages.add("My name is Micayla.");
-    micaylaMessages.add("I am from the Bay Area.");
-    micaylaMessages.add("I live in New York City.");
+  ArrayList<String> comments = new ArrayList<>();
 
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {    
     // Converts ArrayList to a JSON string using Gson.
-    String json = convertToJsonUsingGson(micaylaMessages);
+    String json = convertToJsonUsingGson(comments);
 
     // Sends response as JSON.
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
 
-  private static String convertToJsonUsingGson(ArrayList<String> messages) {
+  private static String convertToJsonUsingGson(ArrayList<String> comments) {
     Gson gson = new Gson();
-    String json = gson.toJson(messages);
+    String json = gson.toJson(comments);
     return json;
   }
 
-}
+@Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String newComment = getParameter(request, "text-input", "");
+
+    // Add input to ArrayList.
+    comments.add(newComment);
+
+    // Creates an Entity with a kind of Task and stores it in taskEntity variable.
+    Entity taskEntity = new Entity("Task");
+
+    // Adds two properties to the taskEntity entity
+    taskEntity.setProperty("new comment", newComment);
+
+    // Creates an instance of the DatastoreService class.
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    // Store entity by passing it into datastore.put()
+    datastore.put(taskEntity);
+
+    // Gives a PreparedQuery instance that contains all of the entities in Datastore with that kind
+    Query query = new Query("Task").addSort("new comment", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+    for (Entity entity : results.asIterable());
```

Comment:
Querying the datastore should go into doGet(). But also notice that the for loop here (line 76) isn't doing anything with the data returned from datastore.